include/openssl/kdf.h

```diff
@@ -115,61 +115,41 @@ void EVP_KDF_names_do_all(const EVP_KDF *kdf,
 # define EVP_PKEY_HKDEF_MODE_EXPAND_ONLY        \
             EVP_KDF_HKDF_MODE_EXPAND_ONLY
 
-# define EVP_PKEY_CTX_set_tls1_prf_md(pctx, md) \
-            EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
-                              EVP_PKEY_CTRL_TLS_MD, 0, (void *)(md))
+int EVP_PKEY_CTX_set_tls1_prf_md(EVP_PKEY_CTX *ctx, const EVP_MD *md);
```

Comment:
Should the EVP_PKEY_CTX wrappers for KDFs be deprecated or at least internal?  This isn't an API we want to promote and the PKEY KDF APIs are **more** appealing than the direct KDF API because of the public helper functions.

I know, add helpers to KDF.

Reply:
Possibly. Although I'd rather not do deprecation as part of this PR. If we're going to do it, I'd rather deprecate separately.

Reply:
Fair enough, later is fine.